### ISSUE TITLE
openjdk8 no longer builds

### ISSUE THREAD

**Issue by purefn** [NixOS/nixpkgs#7898]
```
building path(s) ‘/nix/store/qzfib69wvshrh6kg0d1c827kywffm0kx-openjdk-8u40b25-jre’, ‘/nix/store/s4qm80fb8wkbvwxn376a6wn2i6vxlgws-openjdk-8u40b25’
unpacking sources
unpacking source archive /nix/store/9xdgd07spcl3np3mn106wq7nn8wni176-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/2dsrxpkd7bm6sijapgb7rdyafvylh4xz-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/8fycbdaclxadd43xg66jz7nxa2j291q7-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/q4qa6ad7m1akblbzq26k21vn7475x08h-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/kfyiyr2sd9lckf1wgfig8bq7vdh4xd3m-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/v3nl14930w91ih2psc8sygqdkcmczgpf-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/b1qcsq1jpj26bkp2cm49499d9ri95awa-jdk8u40-b25.tar.gz
unpacking source archive /nix/store/l7dnasi93bcpqq0mk9lw5akpf3v8xgnz-jdk8u40-b25.tar.gz
source root is jdk8u40-jdk8u40-b25
patching sources
applying patch /nix/store/2x6v46j158y86yhpzwhywjc0x7wgn84p-fix-java-home.patch
can't find file to patch at input line 4
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -ru -x '*~' openjdk-orig/hotspot/src/os/linux/vm/os_linux.cpp openjdk/hotspot/src/os/linux/vm/os_linux.cpp
|--- openjdk-orig/hotspot/src/os/linux/vm/os_linux.cpp	2013-09-06 20:22:03.000000000 +0200
|+++ openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2014-01-24 22:44:08.223857012 +0100
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
1 out of 1 hunk ignored
builder for ‘/nix/store/1xyz1zg5n4bmfl3bxyjjg9qh7hzq0hmx-openjdk-8u40b25.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/86g8lgh0aj1l9f15p14rhxdqay5nr05r-system-path.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/kpb9sfwkjjnxk5a0jps95zcw9bnyj01n-nixos-15.06pre62649.f93a8ee.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/kpb9sfwkjjnxk5a0jps95zcw9bnyj01n-nixos-15.06pre62649.f93a8ee.drv’ failed
```

**Comment by edwtjo** [comment]
I'll look into it

**Comment by edwtjo** [comment]
gnupatch feature, fixed in e697219d3d5eae7718eceb5e30bda6765a5b170d

**Comment by purefn** [comment]
Great, thank you!
